tlv.go

```diff
@@ -0,0 +1,153 @@
+// Type-Length-Value splitting and parsing for proxy protocol V2
+// See spec https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt sections 2.2 to 2.7 and
+// Amazon's extension https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-target-groups.html#proxy-protocol
+
+package proxyproto
+
+import (
+	"encoding/binary"
+	"errors"
+	"io"
+	"io/ioutil"
+)
+
+const (
+	// Section 2.2
+	PP2_TYPE_ALPN           PP2Type = 0x01
+	PP2_TYPE_AUTHORITY              = 0x02
+	PP2_TYPE_CRC32C                 = 0x03
+	PP2_TYPE_NOOP                   = 0x04
+	PP2_TYPE_SSL                    = 0x20
+	PP2_SUBTYPE_SSL_VERSION         = 0x21
+	PP2_SUBTYPE_SSL_CN              = 0x22
+	PP2_SUBTYPE_SSL_CIPHER          = 0x23
+	PP2_SUBTYPE_SSL_SIG_ALG         = 0x24
+	PP2_SUBTYPE_SSL_KEY_ALG         = 0x25
+	PP2_TYPE_NETNS                  = 0x30
+	// Section 2.2.7, reserved types
+	PP2_TYPE_MIN_CUSTOM     = 0xE0
+	PP2_TYPE_MAX_CUSTOM     = 0xEF
+	PP2_TYPE_MIN_EXPERIMENT = 0xF0
+	PP2_TYPE_MAX_EXPERIMENT = 0xF7
+	PP2_TYPE_MIN_FUTURE     = 0xF8
+	PP2_TYPE_MAX_FUTURE     = 0xFF
+	// Amazon's extension
+	PP2_TYPE_AWS            = 0xEA
+	PP2_SUBTYPE_AWS_VPCE_ID = 0x01
+)
+
+var (
+	ErrTruncatedTLV    = errors.New("Truncated TLV")
```

Comment:
Per Go standards, the first letter of an error string should not be capitalized ([see here](https://github.com/golang/go/wiki/CodeReviewComments#error-strings)).

Reply:
The existing lib has all errors captialized, so I choose to match the surrounding lib's code.  Eg [	ErrCantReadProtocolVersionAndCommand    = errors.New("Can't read proxy protocol version and command")](https://github.com/elastic/go-proxyproto/pull/1/files#diff-5e68413c67a51bcc7bf6e42ec37bde13L19).  Changing the rest of the code would be an API break, making this less likely to move upstream.